### PR TITLE
Update testing project for .NET Core

### PR DESCRIPTION
**Pull request by dbelcham**
This updates the test project to compile in .NET Core and output the files needed for the build server.

### CODE REVIEW

File: src/Tests/Tests.csproj [hunk 1]
```diff
@@ -7,11 +7,11 @@
     <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="ObjectApproval" Version="1.*" />
+    <PackageReference Include="ObjectApproval" Version="1.*" NoWarn="NU1701"/>
     <PackageReference Include="NUnit" Version="3.7.1" />
-    <PackageReference Include="NUnit3TestAdapter" Version="3.8.0-alpha1" />
-    <PackageReference Include="ApprovalTests" Version="3.*" />
-    <PackageReference Include="ApprovalUtilities" Version="3.*" />
+    <PackageReference Include="NUnit3TestAdapter" Version="3.8.0-alpha1" NoWarn="NU1701"/>
```
Comment (timbussmann) [inline src/Tests/Tests.csproj]: why does this package has a NoWarn?

**Comment by dbelcham** [comment]
@Particular/nservicebus-maintainers the last changes fixed all the build issues around .NET Core so this is good to go

File: src/Tests/ArrayTests.cs [hunk 2]
```diff
@@ -2,17 +2,16 @@
...
 using NServiceBus.Newtonsoft.Json;
 using NUnit.Framework;
-using ObjectApproval;
 
 [TestFixture]
 public class ArrayTests
 {
     string typeName = $"{typeof(ArrayMessage).FullName}, {typeof(ArrayMessage).Assembly.GetName().Name}";
-
+
```
Comment (bording) [inline src/Tests/ArrayTests.cs]: 🔥 trailing whitespace

File: src/Tests/Tests.csproj [hunk 3]
```diff
@@ -1,21 +1,22 @@
...
-    <PackageReference Include="PublicApiGenerator" Version="6.1.0-*" />
     <PackageReference Include="NUnit" Version="3.7.1" />
     <PackageReference Include="NUnit3TestAdapter" Version="3.8.0-alpha1" />
-    <PackageReference Include="ApprovalTests" Version="3.*" />
-    <PackageReference Include="ApprovalUtilities" Version="3.*" />
+    <PackageReference Include="ApprovalTests" Version="3.*" NoWarn="NU1701" />
+    <PackageReference Include="ApprovalUtilities" Version="3.*" NoWarn="NU1701" />
     <PackageReference Include="Mono.Cecil" Version="0.10.0-*" />
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.*" />
+    <PackageReference Include="Newtonsoft.Json" Version="[10.0.3, 11.0.0)" />
```
Comment (bording) [inline src/Tests/Tests.csproj]: Why was this required? I would have expected that to come in transitively from the project reference.

File: src/Tests/TestApprover.cs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                sourcePath = Path.Combine(assemblyDir, "..", "..", "..", "ApprovalFiles");
+            }
+
+            public override string SourcePath => sourcePath;
+
+            readonly string sourcePath;
+        }
+    }
+
+    public static class ObjectApprover
```
Comment (bording) [inline src/Tests/TestApprover.cs]: Not that it really matters, but this doesn't really need to be public.

File: src/Tests/TestApprover.cs [hunk 5]
```diff
@@ -0,0 +1,74 @@
+﻿namespace NServiceBus.Core.Tests
```
Comment (bording) [inline src/Tests/TestApprover.cs]: None of the other classes in the test project have a namespace, so this could be removed, which would let you remove the other new using statements...

File: src/Tests/Tests.csproj [hunk 6]
```diff
@@ -1,21 +1,22 @@
...
   <ItemGroup>
-    <PackageReference Include="ObjectApproval" Version="1.*" />
-    <PackageReference Include="PublicApiGenerator" Version="6.1.0-*" />
     <PackageReference Include="NUnit" Version="3.7.1" />
     <PackageReference Include="NUnit3TestAdapter" Version="3.8.0-alpha1" />
-    <PackageReference Include="ApprovalTests" Version="3.*" />
-    <PackageReference Include="ApprovalUtilities" Version="3.*" />
+    <PackageReference Include="ApprovalTests" Version="3.*" NoWarn="NU1701" />
+    <PackageReference Include="ApprovalUtilities" Version="3.*" NoWarn="NU1701" />
     <PackageReference Include="Mono.Cecil" Version="0.10.0-*" />
```
Comment (bording) [inline src/Tests/Tests.csproj]: If this is only needed for PublicApiGenerator, it could become a net452-only dependency.
